src/eos-shard-writer.c

```diff
@@ -95,29 +158,64 @@ eos_shard_writer_class_init (EosShardWriterClass *klass)
 static void
 eos_shard_writer_init (EosShardWriter *self)
 {
-  self->entries = g_array_new (FALSE, TRUE, sizeof (struct eos_shard_writer_record_entry));
-  g_array_set_clear_func (self->entries, (GDestroyNotify) eos_shard_writer_record_entry_clear);
-}
+  cpool_init (&self->cpool);
 
-static void
-blob_entry_init (struct eos_shard_writer_blob_entry *blob)
-{
-  blob->content_type = g_strdup ("");
+  self->blobs = g_array_new (FALSE, TRUE, sizeof (struct eos_shard_writer_blob_entry));
+  g_array_set_clear_func (self->blobs, (GDestroyNotify) eos_shard_writer_blob_entry_clear);
+
+  self->records = g_array_new (FALSE, TRUE, sizeof (struct eos_shard_writer_record_entry));
+  g_array_set_clear_func (self->records, (GDestroyNotify) eos_shard_writer_record_entry_clear);
 }
 
-static void
-record_entry_init (struct eos_shard_writer_record_entry *record)
+/**
+ * eos_shard_writer_add_blob:
+ * @self: an #EosShardWriter
+ * @name: the name of the blob to store.
+ * @file: a file of contents to write into the shard
+ * @content_type: (allow-none): The MIME type of the blob. Pass %NULL to
+ *   autodetect using Gio.
+ * @flags: flags about how the data should be stored in the file
+ *
+ * Adds the blob at the specified file path to the file.
```

Comment:
a little confusingly worded, suggest `Adds the blob at the specified file path to the shard.`